### PR TITLE
Include missing functions from llvm's compiler-rt into Jitllvm

### PR DESCRIPTION
**Pull request by aguinet**
It's still a hack to fix the "divsion" problem with LLVM, but seems to work under Linux! (haven't tried under Windows yet but hopefully it works)

### CODE REVIEW

**Comment by serpilliere** [comment]
Hi @aguinet !
Sorry for the delay.
Thanks a lot for the idea: I tested on archlinux, we need one more function (divti3)
I did a PR on your PR to fix this, but maybe you prefer

**Comment by aguinet** [comment]
I've squash your commit into mine.
It works for Linux, but IIRC under Windows the hack does not work.

The main remaining issue is how to have distutils build a shared library in a portable way. We might also fix this by launching command line by hand to generate this .so/.dll ...

**Comment by serpilliere** [comment]
Can you rebase on master @aguinet ? I fixed the tipo erros in codespell to pass travis tests.
(sorry for that)

**Comment by aguinet** [comment]
Done!

**Comment by serpilliere** [comment]
Hope the Travis Gods will be on our side.

**Comment by serpilliere** [comment]
The Travis Gods are pleased with your sacrifice. 
But maybe it's time to remove Win32 support.

**Comment by serpilliere** [comment]
I am pushing a dummy PR to see if the windows 32 bit tests are due to this PR or if it's a coincidence.

**Comment by aguinet** [comment]
So I will squash these commits into one!

About Windows 32, I will try on my local VM, but it seems to be a bigger issue.

There is also this https://ci.appveyor.com/project/cea-sec/miasm/builds/33068672/job/701prnol8kpicw4d, with an issue on dse_crackme.py, that can't delete a temporary file. Maybe a race between two tests?

**Comment by serpilliere** [comment]
There is an issue on dse which appears at *runtime* randomly, but in most cases, the *build* is ok on appveyor.
See this #1235 I just did: https://ci.appveyor.com/project/cea-sec/miasm/builds/33069250
The windows tests passed on windows 32 bit.

But it seems your PR broke the test at *build time*, right?

**Comment by aguinet** [comment]
Yes the issue on win 32 bits is at build time. I seem to be able to reproduce this locally. I'm looking into it.

**Comment by aguinet** [comment]
So the issue under 32 bits is that clang still build 64 bits binaries. Indeed, the choice is normally made with the VS environment that is choosen. I will try and fix that properly :)

**Comment by aguinet** [comment]
Other problem is that uint128_t does not exist for Windows 32-bit, even with Clang. IIRC it is the same for Linux 32 bits. I guess we have to disable the mn_div test for 32-bit builds!

**Comment by aguinet** [comment]
Fixed by forcing the compilation with clang only for 64-bit Windows systems. I also disabled the mn_div.py test if we are not running on a 64-bit system.

**Comment by aguinet** [comment]
All green @serpilliere !

File: setup.py [hunk 1]
```diff
@@ -36,6 +40,48 @@ def run(self):
         ]
         return install_data.run(self)
 
+def win_get_llvm_reg():
+    REG_PATH = "SOFTWARE\\LLVM\\LLVM"
+    try:
+      return winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, REG_PATH, 0, winreg.KEY_READ | winreg.KEY_WOW64_32KEY)
+    except FileNotFoundError:
+      pass
+    return winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, "SOFTWARE\\LLVM\\LLVM", 0, winreg.KEY_READ)
```
Comment (serpilliere) [inline setup.py:49]: Maybe we can use the variable `REG_PATH` here?

Comment (aguinet) [reply setup.py:49]: Yes, copy/paste is a terrible feature :)

**Comment by serpilliere** [comment]
Thanks a lot for the fix @aguinet !

**Pull request merged**